# BPAGS: A web Application for Bacteriocin Prediction

If you find our web application (https://shiny.tricities.wsu.edu/bacteriocin-prediction/) useful, please cite the following paper:

â€¢ Akhter, S. and Miller, J.H., BPAGS: A web application for bacteriocin prediction via feature evaluation using alternating decision tree, genetic algorithm, and linear support vector classifier. Frontiers in Bioinformatics, 3, p.1284705.


Scripts and Datasets:

**feature_extract.R** - To extract features from protein sequences. 

**pearson's_corr.R** - To perform Pearson correlation analysis.

**GA.R** - To perform genetic algorithm using Support Vector Machine and Random Forest.

**lib_ga.R** - To extract features from the genetic algorithm. 

**ADT.R** - To perform alternating decision tree analysis.

**ml_svm.R** - To implement Support Vector Machine.

**ml_rf.R** - To implement the Random Forest model.

**ml_dt.R** - To implement the Decision Tree model.

**ml_lr.R** - To implement the logistic regression model.

**ml_knn.R** - To implement the k-nearest neighbors model.

**ml_gnb.R** - To implement the Gaussian Naive Bayes model.

**positiveBac.faa and negativeBac.faa** - Bacteriocin (positive) and non-bacteriocin (negative) FASTA sequences.


